addons/website/views/website_templates.xml

```diff
@@ -170,7 +170,7 @@
     </xpath>
 
     <xpath expr="//header/t[@t-cache]" position="attributes">
-        <attribute name="t-cache">xmlid,website</attribute>
+        <attribute name="t-cache">xmlid,website,website.is_public_user()</attribute>
```

Comment:
`t-cache` is already too obscur by nature, let's add a quick comment about why this part of the `t-cache` tuple exists, something like
```py
# website.is_public_user() is needed for menu having a page with restricted visibility (only shown for logged in user), public user and logged in user can't share the menu cache
```